gobblin-core/src/main/java/gobblin/writer/objectstore/ObjectStoreClient.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.writer.objectstore;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import com.typesafe.config.Config;
+
+/**
+ * A client interface to interact with an object store. Supports basic operations like put,delete and get.
+ */
+public interface ObjectStoreClient {
+
+  /**
+   * Puts an object in <code>objectStream</code> to the store. Returns the id of the object created.
+   *
+   * @param objectStream to put in the store
+   * @param putConfig additional config if any (User metadata, put options etc.)
+   * @return the id of newly created object
+   * @throws IOException if put failed
+   */
+  public String put(InputStream objectStream, Config putConfig) throws IOException;
+
+  /**
+   * Delete an object with <code>objectId</code> in the store. Operation is a noop if object does not exist
+   *
+   * @param objectId to delete
+   * @param deletetConfig additional config if any
+   * @throws IOException if delete failed
+   */
+  public void delete(String objectId, Config deletetConfig) throws IOException;
+
+  /**
+   * Get metadata associated with an object
+   * @param objectId for which metadata is retrieved
+   * @return object metadata
+   * @throws IOException if get object metadata fails
+   */
+  public Config getObjectProps(String objectId) throws IOException;
```

Comment:
I can also see a setObjectProps command too. For example, http://developer.openstack.org/api-ref/object-storage/index.html